optuna/samplers/base.py

```diff
@@ -33,4 +101,5 @@ def sample(self, storage, study_id, param_name, param_distribution):
 
         """
 
-        raise NotImplementedError
+        tpe = optuna.samplers.TPESampler()
+        return tpe.sample_independent(study, trial, param_name, param_distribution)
```

Comment:
It looks current implementation of `BaseSampler` behaves the same with `TPESampler`. I have a strange feeling in this behavior although I think `TPESampler` is better than `RandomSampler` in most of the cases.

I guess the reason why you call `TPESampler` in this line is to be able to define third party's sampler more simply. People who want to implement GPEI sampler don't need to define `sample_independent` method. But if it will appear new SoTA optimization method in the future, you might think use it. It will be braking changes for third party's sampler which doesn't override `sample_independent` method.

Reply:
I think your option makes sense.
But, I would also like to hear other people's opinions.
What do you think? > @g-votte @toshihikoyanase

Reply:
We decided, at least in the first version, to remove the default implementations from `BaseSampler` (commit: (https://github.com/pfnet/optuna/pull/380/commits/c66ec702c761f16b99e4a3fa97a700dd6c4cb5e4) because it turned out that it is difficult to provide default implementations that cover all edge cases well.

Thank you for your useful comment.